prow/git/v2/interactor.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+// Interactor knows how to publish local work to a remote
+type Interactor interface {
+	Directory() string
+	Clean() error
+	MirrorClone() error
+	Clone(from string) error
+	Checkout(commitlike string) error
+	RevParse(commitlike string) (string, error)
+	BranchExists(branch string) bool
+	CheckoutNewBranch(branch string) error
+	Merge(commitlike string) (bool, error)
```

Comment:
As mentioned in the original PR, I think everything that merges should be done as part of the `Publisher` interface so it can use the author configuration.

As an aside, what do we gain from splitting the interactor and publisher interfaces and then exposing an union interface over just exposing one interface for everything?

Reply:
Thematic separation is all, same as for the k8s clientsets or the GitHub client